libcaf_core/caf/intrusive/linked_list.hpp

```diff
@@ -50,9 +50,7 @@ class linked_list {
   linked_list() noexcept = default;
 
   linked_list(linked_list&& other) noexcept {
-    if (other.empty()) {
-      init();
-    } else {
+    if (!other.empty()) {
```

Comment:
If you do `my_list = linked_list{}`, then `my_list` should be empty. This implementation turns it into a no-op.

Reply:
Not sure I understand this fully. 
`my_list = linked_list{}` calls the `opterator=` for `my_list`, which will call `init` since the right side is empty.
The `linked_list{}` call is a default ctor. 
If we would write `auto my_list = std::move(some_empty_list);` since we don't have an initializer list, the values for head and tail will be default initialized like in the class body to the addresses of one another, like they would be when calling `init`, and the ctor body won't touch them afterwards. 
That would be my understanding, could you point me in the right direction if I got that wrong?

Reply:
Sorry, I've confused the move-constructor with the copy-assignment. :)